presto-main/src/main/java/com/facebook/presto/operator/scalar/MapConstructor.java

```diff
@@ -121,55 +120,52 @@ public BuiltInScalarFunctionImplementation specialize(BoundVariables boundVariab
         MethodHandle keyBlockEquals = compose(keyNativeEquals, nativeValueGetter(keyType), nativeValueGetter(keyType));
         MethodHandle keyIndeterminate = functionAndTypeManager.getBuiltInScalarFunctionImplementation(
                 functionAndTypeManager.resolveOperator(INDETERMINATE, fromTypeSignatures((keyType.getTypeSignature())))).getMethodHandle();
-        MethodHandle instanceFactory = constructorMethodHandle(State.class, MapType.class).bindTo(mapType);
 
         return new BuiltInScalarFunctionImplementation(
                 false,
                 ImmutableList.of(
                         valueTypeArgumentProperty(RETURN_NULL_ON_NULL),
                         valueTypeArgumentProperty(RETURN_NULL_ON_NULL)),
-                METHOD_HANDLE.bindTo(mapType).bindTo(keyBlockEquals).bindTo(keyBlockHashCode).bindTo(keyIndeterminate),
-                Optional.of(instanceFactory));
+                METHOD_HANDLE.bindTo(mapType).bindTo(keyType).bindTo(valueType).bindTo(keyBlockEquals).bindTo(keyBlockHashCode).bindTo(keyIndeterminate),
+                Optional.empty());
     }
 
     @UsedByGeneratedCode
     public static Block createMap(
             MapType mapType,
+            Type keyType,
```

Comment:
You can just use `mapType.getKeyType()` and `mapType.getValueType()` for key and value types.

Reply:
We want to bindTo hoping it will inline and move the deadcode. Not sure if it will with mapType.getKeyType()